### DISCUSSION TITLE
[error] Ekka(AutoCluster): Discovery error: {404

### DISCUSSION THREAD

**Discussion by yukuochang** [emqx/emqx#9507] [General]
version 4.3.5 
Helm chart install
<img width="1440" alt="图片" src="https://user-images.githubusercontent.com/55518766/206356169-c23f491e-4ec7-47fd-9abf-a1560cd7a09c.png">
<img width="1229" alt="图片" src="https://user-images.githubusercontent.com/55518766/206356238-f37e39d3-e197-48bd-b862-71a23fd60b94.png">
<img width="1057" alt="图片" src="https://user-images.githubusercontent.com/55518766/206356645-1023790f-60b5-4a41-a9a2-bb2fde4d423f.png">
<img width="872" alt="图片" src="https://user-images.githubusercontent.com/55518766/206357753-90746895-c5d1-4f64-9cb5-669b28cf5021.png">
<img width="1147" alt="图片" src="https://user-images.githubusercontent.com/55518766/206357842-d148a28e-f04e-49ac-aeac-dbbca4497d6c.png">
<img width="861" alt="图片" src="https://user-images.githubusercontent.com/55518766/206361580-ad2395a9-ad33-48a6-8de4-a5c7249ff761.png">

[error] Ekka(AutoCluster): Discovery error: {404,"{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"endpoints \\\"emqx-headless\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"emqx-headless\",\"kind\":\"endpoints\"},\"code\":404}\n"}

**Comment by zmstone** [comment]
Hi
404 code implies that the container does not have permission to query the k8s API server.
Alternatives are to use DNS or etcd discovery.
